tests/test_curves.py

```diff
@@ -844,3 +844,48 @@ def test_compute_ndarray_error(self):
         c = SplineXYZCurve(X=R * np.cos(phi), Y=R * np.sin(phi), Z=np.zeros_like(phi))
         with pytest.raises(TypeError):
             c.compute("length", grid=np.linspace(0, 1, 10))
+
+    @pytest.mark.unit
+    def test_discontinuous_knots(self):
+        """Test with discontinuous knots."""
+        discontinuous_indices = [0, 500]
+        R = 2
+        phi = 2 * np.pi * np.linspace(0, 1, 1001, endpoint=True) ** 2
+
+        def test(method):
+            discontinuous = SplineXYZCurve(
+                X=R * np.cos(phi),
+                Y=R * np.sin(phi),
+                Z=np.zeros_like(phi),
+                knots="arclength",
+                discontinuous_indices=discontinuous_indices,
+                method=method,
+            )
+
+            continuous = SplineXYZCurve(
+                X=R * np.cos(phi),
+                Y=R * np.sin(phi),
+                Z=np.zeros_like(phi),
+                knots="arclength",
+                discontinuous_indices=None,
+                method=method,
+            )
+
+            assert discontinuous.method == method
+            assert continuous.method == method
+
+            discont_curvature = discontinuous.compute("curvature")["curvature"]
+            cont_curvature = continuous.compute("curvature")["curvature"]
+
+            return discont_curvature, cont_curvature
+
+        discont_curvature, cont_curvature = test("linear")
+        np.testing.assert_allclose(discont_curvature, cont_curvature, rtol=1e-3)
```

Comment:
Is it worth testing any other compute quantities besides just curvature?

Reply:
I could add one for length pretty length pretty easily but not sure what I would add for torsion. I guess checking that the torsion is zero for each of them could work?